rlog_test.go

```diff
@@ -0,0 +1,241 @@
+func setup() rlogConfig {
+	if fixedLogfileName {
+		logfile = "/tmp/rlog-test.log"
+	} else {
+		logfile = fmt.Sprintf("/tmp/rlog-test-%d.log", time.Now().UnixNano())
+	}
+	os.Remove(logfile)
+
+	return rlogConfig{
+		logLevel:       "",
+		traceLevel:     "",
+		logTimeFormat:  "",
+		logFile:        logfile,
+		logStream:      "NONE",
+		logTimeDate:    false,
+		showCallerInfo: false,
+	}
+}
+
+// cleanup is called at the end of each test.
+func cleanup() {
+	if removeLogfile {
+		os.Remove(logfile)
+	}
+}
+
+// fileMatch compares entries in the logfile with expected entries provided as
+// a list of strings (one for each line)
+func fileMatch(t *testing.T, checkLines []string, timeLayout string) {
+	file, err := os.Open(logfile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	i := 0
+	for scanner.Scan() {
+		line := scanner.Text()
+		if timeLayout != "" {
+			dateTime := line[:len(timeLayout)]
+			line = line[len(timeLayout)+1:]
+			_, err := time.Parse(timeLayout, dateTime)
+			if err != nil {
+				t.Fatal(err)
+				t.Fatalf("Incorrect date/time format.\nSHOULD: %s\nIS:     %s\n", timeLayout, dateTime)
+			}
+		}
+		if i >= len(checkLines) {
+			t.Fatal("Not enough lines provided in checkLines.")
+		}
+		if line != checkLines[i] {
+			t.Fatalf("Log line %d does not match check line.\nSHOULD: %s\nIS:     %s\n", i, checkLines[i], line)
+		}
+		i++
+	}
+	if len(checkLines) > i {
+		t.Fatalf("Only %d of %d checklines found in output file.", i, len(checkLines))
+	}
+	if i == 0 {
+		t.Fatal("No input scanned")
+	}
+}
+
+func TestLogLevels(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "DEBUG"
+	Initialize(conf)
+
+	Debug("Test Debug")
+	Info("Test Info")
+	Warn("Test Warning")
+	Error("Test Error")
+	Critical("Test Critical")
+
+	checkLines := []string{
+		"DEBUG    : Test Debug",
+		"INFO     : Test Info",
+		"WARN     : Test Warning",
+		"ERROR    : Test Error",
+		"CRITICAL : Test Critical",
+	}
+	fileMatch(t, checkLines, "")
+}
+
+func TestLogLevelsLimited(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "WARN"
+	conf.traceLevel = "3"
+	Initialize(conf)
+
+	Debug("Test Debug")
+	Info("Test Info")
+	Warn("Test Warning")
+	Error("Test Error")
+	Critical("Test Critical")
+	Trace(1, "Trace 1")
+	Trace(2, "Trace 2")
+	Trace(3, "Trace 3")
+	Trace(4, "Trace 4")
+	checkLines := []string{
+		"WARN     : Test Warning",
+		"ERROR    : Test Error",
+		"CRITICAL : Test Critical",
+		"TRACE(1) : Trace 1",
+		"TRACE(2) : Trace 2",
+		"TRACE(3) : Trace 3",
+	}
+	fileMatch(t, checkLines, "")
+}
+
+func TestLogFormatted(t *testing.T) {
+	conf := setup()
+	defer cleanup()
+
+	conf.logLevel = "DEBUG"
+	conf.traceLevel = "1"
+	Initialize(conf)
+
+	Debugf("Test Debug %d", 123)
+	Infof("Test Info %d", 123)
+	Warnf("Test Warning %d", 123)
+	Errorf("Test Error %d", 123)
+	Criticalf("Test Critical %d", 123)
+	Tracef(1, "Trace 1 %d", 123)
```

Comment:
adding trace(2) here would help in negative testing, above negative is for trace, this would be for tracef.